lib/datadog/tracing/sampling/span/sampler.rb

```diff
@@ -50,11 +50,18 @@ def initialize(rules = [])
           def sample!(trace_op, span_op)
             return if trace_op.sampled? && trace_op.priority_sampled?
 
-            # Return as soon as one rule matches
-            @rules.any? do |rule|
-              rule.sample!(span_op) != :not_matched
+            # Applies the first matching rule
+            final_decision = nil
+            @rules.each do |rule|
+              decision = rule.sample!(span_op)
+              if decision != :not_matched
+                final_decision = decision
+                break
+              end
             end
```

Comment:
The final sampling decision wasn't captured in this `any?` loop before, but it's now important to know when to tag the `sampling_mechanism`.